src/main/java/slimeknights/tconstruct/smeltery/TinkerSmeltery.java

```diff
@@ -92,12 +104,51 @@
+  public static final ItemObject<Item> axeHeadCast = ITEMS.register("axe_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> scytheHeadCast = ITEMS.register("scythe_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> sharpeningKitCast = ITEMS.register("sharpening_kit_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> swordBladeCast = ITEMS.register("sword_blade_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> largeSwordBladeCast = ITEMS.register("large_sword_blade_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> knifeBladeCast = ITEMS.register("knife_blade_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> bowLimbCast = ITEMS.register("bow_limb_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> hammerHeadCast = ITEMS.register("hammer_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> excavatorHeadCast = ITEMS.register("excavator_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> wideGuardCast = ITEMS.register("wide_guard_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> handGuardCast = ITEMS.register("hand_guard_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> crossGuardCast = ITEMS.register("cross_guard_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> gearCast = ITEMS.register("gear_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> broadAxeHeadCast = ITEMS.register("broad_axe_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> shovelHeadCast = ITEMS.register("shovel_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> arrowHeadCast = ITEMS.register("arrow_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> kamaHeadCast = ITEMS.register("kama_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> panHeadCast = ITEMS.register("pan_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> signHeadCast = ITEMS.register("sign_head_cast", SMELTERY_PROPS);
+  public static final ItemObject<Item> toughBindingCast = ITEMS.register("tough_binding_cast", SMELTERY_PROPS);
+
 
+  /*
+   * Recipe
+   */
+  public static final RecipeSerializerObject<AbstractCastingRecipe.Serializer<CastingBasinRecipe>> basinRecipeSerializer = RECIPE_SERIALIZERS.register("casting_basin", () -> new AbstractCastingRecipe.Serializer<>(CastingBasinRecipe::new));
+  public static final RecipeSerializerObject<AbstractCastingRecipe.Serializer<CastingTableRecipe>> tableRecipeSerializer = RECIPE_SERIALIZERS.register("casting_table", () -> new AbstractCastingRecipe.Serializer<>(CastingTableRecipe::new));
+  public static final IRecipeType<CastingBasinRecipe> basinRecipeType = IRecipeType.register(Util.resource("casting_basin"));
+  public static final IRecipeType<CastingTableRecipe> tableRecipeType = IRecipeType.register(Util.resource("casting_table"));
```

Comment:
You can make these just `IRecipeType<AbstractCastingRecipe>`. Being less specific will not hurt anything as the table and basin classes do not add anything unique. Same for the serializers. Lets you fix odd casts calling it